src/core_plugins/kibana/public/management/react/index-pattern/index-pattern-list/index-pattern-list.container.js

```diff
@@ -0,0 +1,41 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { IndexPatternList as IndexPatternListComponent } from './index-pattern-list.component';
+
+import {
+  TableProps,
+} from 'plugins/kibana/management/react/lib/table_props';
+
+import {
+  fetchIndexPatterns,
+} from 'plugins/kibana/management/react/store/actions';
+
+import {
+  getIndexPatterns,
+} from 'plugins/kibana/management/react/store/reducers';
+
+const filter = (item, filterBy) => !filterBy['attributes.title'] || item.attributes.title.includes(filterBy['attributes.title']);
+
+const IndexPatternList = connect(
+  state => ({ indexPatterns: getIndexPatterns(state) }),
+  { fetchIndexPatterns },
+)(class extends Component {
+  componentWillMount() {
+    this.props.fetchIndexPatterns();
+  }
+  render() {
+    const { indexPatterns } = this.props;
+    return (
+      <TableProps
```

Comment:
I've thought about this some more, and it really does feel like things such as `TableProps` and `TabProps` should be _owned_ by the component (`IndexPatternListComponent` in this case), instead of provided to it.

One solution could be to just redefine `IndexPatternListComponent` so that it's a class:

```jsx
class IndexPatternList extends Component {
  componentWillMount() {
    this.props.fetchIndexPatterns();
  }
  render() {
    const { indexPatterns } = this.props;
    return (
      <TableProps
        sortBy="attributes.title"
        items={indexPatterns}
        filters={[filter]}
        render={({
          items,
          page,
          perPage,
          sortBy,
          sortAsc,
          numOfPages,
          filterBy,
          items,
          setPage,
          setPerPage,
          setFilterBy,
          setSortBy,
        }) => (
          // Everything from the current IndexPatternList goes here
        )}
      />
    );
  }
}
```

Though personally, I still find imperative interfaces a bit simpler to follow for this kind of thing, like in my `Tabs` example.